atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -208,26 +231,25 @@ boolean runOnce() {
                 });
     }
 
-    /**
-     * Check whether the table being swept was dropped. If so, stop sweeping it and move on.
-     *
-     * @return Whether the table being swept was dropped
-     */
-    boolean checkAndRepairTableDrop() {
+    private SweepOutcome determineCauseOfFailure(Exception originalException) {
         try {
             Set<TableReference> tables = specificTableSweeper.getKvs().getAllTableNames();
             Optional<SweepProgress> progress = specificTableSweeper.getTxManager().runTaskReadOnly(
                     specificTableSweeper.getSweepProgressStore()::loadProgress);
+
             if (!progress.isPresent() || tables.contains(progress.get().tableRef())) {
-                return false;
+                log.warn("The background sweep job failed unexpectedly; will retry with a lower batch size...",
+                        originalException);
+                return SweepOutcome.RETRYING_WITH_SMALLER_BATCH;
             } else {
                 specificTableSweeper.getSweepProgressStore().clearProgress();
-                return true;
+                log.info("The table being swept by the background sweeper was dropped, moving on...");
+                return SweepOutcome.TABLE_DROPPED_WHILE_SWEEPING;
             }
-        } catch (RuntimeException e) {
-            log.error("Failed to check whether the table being swept was dropped."
-                    + " Continuing under the assumption that it wasn't...", e);
-            return false;
+        } catch (RuntimeException newE) {
+            log.error("Sweep failed", originalException);
+            log.error("Failed to check whether the table being swept was dropped. Retrying...", newE);
+            return SweepOutcome.ERROR;
```

Comment:
Note that we would RETRYING_WITH_SMALLER_BATCH here too in the past.

Reply:
done